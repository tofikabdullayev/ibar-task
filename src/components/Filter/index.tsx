import React, { FormEvent, useState, ChangeEvent } from 'react';
import { Button, Form, Input } from 'reactstrap';

interface Props {
  onFilter: (filerText: string) => void;
}

const Filter = ({ onFilter }: Props) => {
  const styles = { display: 'flex', margin: '20px 0' };
  const [value, setValue] = useState<string>('');
  const onSubmit = (event: FormEvent<HTMLFormElement>): void => {
    event.preventDefault();
    onFilter(value);
  };
  return (
    <Form style={styles} onSubmit={onSubmit}>
      <Input
        type="text"
        name="filter"
        placeholder="Type to filter"
        value={value}
        onChange={(event: ChangeEvent<HTMLInputElement>) =>
          setValue(event.target.value)
        }
      />
      <Button>Filter</Button>
    </Form>
  );
};

export default Filter;
