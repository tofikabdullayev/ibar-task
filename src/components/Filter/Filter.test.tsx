import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Filter from './index';

configure({ adapter: new Adapter() });
test('renders the component', () => {
  const component = shallow(<Filter onFilter={() => {}} />);
  expect(component).toMatchSnapshot();
});
