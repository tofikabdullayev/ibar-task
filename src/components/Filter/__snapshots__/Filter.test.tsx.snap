// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Form
  onSubmit={[Function]}
  style={
    Object {
      "display": "flex",
      "margin": "20px 0",
    }
  }
  tag="form"
>
  <Input
    name="filter"
    onChange={[Function]}
    placeholder="Type to filter"
    type="text"
    value=""
  />
  <Button
    color="secondary"
    tag="button"
  >
    Filter
  </Button>
</Form>
`;
